query-algebrizer/src/clauses/or.rs

```diff
@@ -271,25 +281,23 @@ impl ConjoiningClauses {
         assert!(or_join.clauses.len() >= 2);
 
         match self.deconstruct_or_join(schema, or_join) {
+            DeconstructedOrJoin::KnownSuccess => {
+                // The pattern came to us empty -- `(or)`. Do nothing.
+                Ok(())
+            },
             DeconstructedOrJoin::KnownEmpty(reason) => {
                 // There were no arms of the join that could be mapped to a table.
-                // The entire `or` can yield no results.
+                // The entire `or`, and thus the CC, cannot yield results.
```

Comment:
Is this true with nested `or`?  With nested `and`, which is written as concatentation in Datomic IIRC?